TypeCobol/Compiler/Diagnostics/CodeElementCheckers.cs

```diff
@@ -201,53 +201,52 @@ public static void OnCodeElement(CallStatement statement, CodeElementsParser.Cal
                 if (context.programNameOrProgramEntryOrProcedurePointerOrFunctionPointerVariable() == null)
                     DiagnosticUtils.AddError(statement, "Empty CALL is not authorized", context.Start);
 
-                foreach (var call in context.callUsingParameters()) CheckCallUsings(statement, call);
+                // Check each parameter of the CALL
+                foreach (var inputParameter in statement.InputParameters) CheckCallUsings(statement, inputParameter);
 
                 if (context.callReturningParameter() != null && statement.OutputParameter == null)
                     DiagnosticUtils.AddError(statement, "CALL .. RETURNING: Missing identifier", context);
             }
         }
 
-        private static void CheckCallUsings(CallStatement statement, CodeElementsParser.CallUsingParametersContext context)
+        private static void CheckCallUsings(CallStatement statement, CallSiteParameter input)
         {
-            foreach (var input in statement.InputParameters)
-            {
-                // TODO#249 these checks should be done during semantic phase, after symbol type resolution
-                // TODO#249 if input is a file name AND input.SendingMode.Value == SendingMode.ByContent OR ByValue
-                //	DiagnosticUtils.AddError(statement, "CALL .. USING: <filename> only allowed in BY REFERENCE phrase", context);
-                bool isFunctionCallResult = input.StorageAreaOrValue != null &&
-                                            input.StorageAreaOrValue.StorageArea is FunctionCallResult;
+            var errorPosition = input.StorageAreaOrValue?.MainSymbolReference?.NameLiteral.Token;
```

Comment:
This means that the error will always be positionned on the argument name, what about ommited parameters ? What about messages that concern the whole arrgument including its sharing mode ?
The current build fails, maybe it is because you changed the error positionning. I think we need to find a better strategy for this part.

Reply:
For the position of the error, if you give it a context, it will take the first token of this context (so it positions the error on the whole argument).
I think that on ommited parameters, the `AddError` method using a context had a `null` returned for first token, as it has now from `errorPosition`.
Looking at the cases in the function, it seems the new position is only clearer every time. As a developper, I would preferer having the error on the parameter rather than on the "BY".

Reply:
To achieve positioning on a whole arg like `by reference length of txt`, we would have to use the constructor of `Position` class, specifying manually start and end so if @rooksdo is ok with positioning only on arg name, it's good for me too.
I'm not sure for ommitted though because, in previous code we would use the first token of the context so that should give the `OMMITTED` keyword and not `null`, could you please check that ?

Reply:
Previously, the diagnostic from `BY VALUE OMITTED` was placed on the `BY`, even on the case where OMITTED was the second parameter by value (`BY VALUE toto OMITTED`).
Now the error is placed on the token `OMITTED`. (See Call.ko.txt)